build/action.yml

```diff
@@ -0,0 +1,45 @@
+name: Build
+description: Build with Maven and upload build artifacts
+inputs:
+  use_artifacts:
+    description: Upload build artifacts
+    default: 'true'
+  artifact_name:
+    description: 'Artifact name. Default: The name of the build and commit hash'
+outputs:
+  artifact_name:
+    description: Artifact name
+    value: ${{ steps.artifact_name.outputs.artifact_name }}
+  mule_artifact_name:
+    description: Mule artifact name
+    value: ${{ steps.artifact_name.outputs.mule_artifact_name }}
+
+runs:
+  using: composite
+  steps:
+    - name: Build with Maven
+      shell: bash
+      run: mvn -B package --file pom.xml -DskipMunitTests
+
+    - name: Stamp artifact file name with commit hash
+      shell: bash
+      id: artifact_name
+      run: |
+        currentArtifactName=$(ls target/*.jar | head -1)
+        commitHash=$(git rev-parse --short "$GITHUB_SHA")
+        newArtifactName=$(ls target/*.jar | head -1 | sed "s/.jar/-$commitHash.jar/g")
+        mv $currentArtifactName $newArtifactName
+        echo "mule_artifact_name=$newArtifactName" >> $GITHUB_OUTPUT
```

Comment:
I'm confuse with the `mule_artifact_name` vs the `artifact_name` 🤔 as I understand, the `mule_artifact_name` seems like a directory, is it @hoangmirs should we change it to another name